socorro/siglists/README.rst

```diff
@@ -0,0 +1,195 @@
+Signatures Utilities Lists
+==========================
+
+This folder contains lists that are used to configure the C signature generation
+process. Each ``.txt`` file contains a list of signatures or regex matching
+signatures, that are used at various steps of our algorithm. Regular expressions
+use the `Python syntax
+<https://docs.python.org/2/library/re.html#regular-expression-syntax>`_.
+
+
+Signature Generation Algorithm
+------------------------------
+
+When generating a C signature, 5 steps are involved.
+
+1. We walk the crashing thread's stack, looking for things that would match the
+   `Signature Sentinels <#signature-sentinels>`_. The first matching element, if
+   any, becomes the top of the sub-stack we'll consider going forward.
+2. We walk the stack, ignoring everything that matches the `Irrelevant
+   Signatures <#irrelevant-signatures>`_. We consider the first non-matching
+   element the top of the new sub-stack.
+3. We rewrite every signature missing symbols that matches the `Trim DLL
+   Signatures <#trim-dll-signatures>`_ to be the module only (the part before
+   the first ``@`` sign). We also merge them so only one of those frames makes
+   it to the final signature.
+4. We accumulate signatures that match the `Prefix Signatures
+   <#prefix-signatures>`_, until something doesn't match.
+5. We normalize each signature we accumulated. Signatures that match the
```

Comment:
As a followup, it would be great to document what "normalize" means here. I was looking at [a bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1397926) this morning that's going to want to add a new normalization rule, and I looked at the siglists doc first before realizing that it wasn't documented there. I can probably write the explanation from reading the current code, or you can take a crack at it and I can edit.

Reply:
Yeah... this is still kind of crap for docs about how it all works.

Bug 1306643 (https://bugzilla.mozilla.org/show_bug.cgi?id=1306643) covers documenting the rules. I definitely want to do that. I was thinking of fixing the docstrings and then running them through autodoc and exposing the `DEFAULT_PIPELINE` for ordering. Or something along those lines that fleshes out the documentation without creating a maintenance burden.

I think this will improve when someone works on that bug.